rocksdb_admin/admin_handler.cpp

```diff
@@ -1127,20 +1127,27 @@ void AdminHandler::async_tm_stopMessageIngestion(
   const auto db_name = request->db_name;
   LOG(ERROR) << "Called stopMessageIngestion for " << db_name;
 
-  auto sync_variable = sync_var_map.find(db_name);
+  {
+    std::lock_guard<std::mutex> lock(sync_var_map_mutex);
+    auto sync_variable = sync_var_map.find(db_name);
 
-  // Verify that messages are being consumed for this partition
-  if (sync_variable == sync_var_map.end()) {
-    e.message = db_name + " consumption already stopped";
-    callback.release()->exceptionInThread(std::move(e));
-    LOG(ERROR) << "Could not find condition variable for " << db_name <<
-                  ". Consumption might have never started or already stopped";
-    return;
+    // Verify that messages are being consumed for this partition
+    if (sync_variable == sync_var_map.end()) {
+      e.message = db_name + " consumption already stopped";
+      callback.release()->exceptionInThread(std::move(e));
+      LOG(ERROR) << "Could not find condition variable for " << db_name <<
+                    ". Consumption might have never started or already stopped";
+      return;
+    }
+
+    // Set atomic flag and notify consuming thread.
+    {
```

Comment:
It would be better to unlock sync_var_map_mutex first then lock the mutex in the sync variable

Reply:
Good catch, this will ensure the mutex is locked for very short durations.